frontend/src/components/navbar/Navbar.tsx

```diff
@@ -7,11 +7,19 @@ import calendar from '../../assets/Calendar.png';
 import circle from '../../assets/Circle.png';
 import work from '../../assets/Work.png';
 import sidebar from '../../assets/Sidebar.png';
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 
 function NavBar() {
   const location = useLocation();
-  const [selectedLink, setSelectedLink] = useState(location.pathname);
+  const [selectedLink, setSelectedLink] = useState(() => {
+    const savedLink = localStorage.getItem('selectedLink');
+    return savedLink ? savedLink : location.pathname;
+  });
+
+  useEffect(() => {
+    localStorage.setItem('selectedLink', selectedLink);
```

Comment:
Why do we need to store it in the localStorage?

Reply:
we are using localstorage to save information of web browser.  because in our website when we are on http://localhost:8000/company this url and refresh the page it shows the Company selected in sidebar but when we are on company details page http://localhost:8000/CompanyDetails/3 here and if we refresh page its not showing Company selected in sidebar. It was showing Home selected. To solve this i have used localstorage.

Reply:
ooh I see, and is `CompanyDetails` a valid path?

Reply:
Yes, its a valid path.